src/main/java/core/basesyntax/strategy/handlers/PurchaseHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy.handlers;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseHandler implements OperationHandler {
+    private final StorageDao storageDao;
+
+    public PurchaseHandler(StorageDao storage) {
+        this.storageDao = storage;
+    }
+
+    @Override
+    public void handle(String productType, int amount) {
+        int newAmount = storageDao.getAmountByProduct(productType) - amount;

```

Comment:
what if we have 0 amount of that product left in the storage, can we still sell it? Let's throw an ex in that case